src/components/AddRecipe.js

```diff
@@ -0,0 +1,257 @@
+                  required
+                />
+              ))}
+                <div className="valid-feedback">Looks good!</div>
+                  </Col>
+                  <Col>
+                  <label
+                  htmlFor="defaultFormRegisterConfirmEx3"
+                  className="grey-text"
+                >
+              </label>
+              {this.state.measurement.map(unit=> (
+                <select
+                  onChange={this.changeHandler}
+                  key={unit.name}
+                  type="select"
+                  id="defaultFormRegisterConfirmEx3"
+                  className="form-control"
+                  name="unit"
+                  placeholder="Unit">
+                  <option value="">Unit</option>
+                  <option value="c(Cup)">c(Cup)</option>
+                  <option value="l(Liter)">l(Liter)</option>
+                  <option value="kg(Kilogram)">kg(Kilogram)</option>
+                  <option value="g(Gram)">g(Gram)</option>
+                  <option value="tsp(Teaspoon)">tsp(Teaspoon)</option>
+                  <option value="dp(Drop)">dp(Drop)</option>
+                  <option value="oz(Ounce)">oz(Ounce)</option>
+                  <option value="Tbsp(Tablespoon)">Tbsp(Tablespoon)</option>
+                </select>
+              ))}
+                  </Col>
+                  <Col>
+                  <label
+                  htmlFor="defaultFormRegisterNameEx"
+                  className="grey-text"
+                >
+              </label>
+              {this.state.measurement.map(ingredient=> (
+                <input type="text"
+                key={ingredient.name}
+                  name="ingredient"
+                  onChange={this.changeHandler}
+                  id="defaultFormRegisterNameEx"
+                  className="form-control"
+                  placeholder="Ingredient"
+                  required
+                />
+              ))}
+                <div className="valid-feedback">Looks good!</div>
+                  </Col>
+              </Row>
+              <h2 className="header1">Add Steps</h2>
+              <Row>
+              <Button className="button-plus" onClick={()=>{
+                        this.setState(previousState=>({steps:[...previousState.steps,
+                         {steps:"", number:previousState.currentStep}],
+                          currentStep: previousState.currentStep++}))}}>
+              <FaPlus />
+              </Button>
+              <Col xs={2}>
+              <label
+                  htmlFor="defaultFormRegisterNameEx"
+                  className="grey-text"
+                >
+              </label>
+              {this.state.steps.map(number=> (
+                <input 
+                key={number.name}
+                  onChange={this.changeHandler}
+                  id="defaultFormRegisterNameEx"
+                  className="form-control"
+                  placeholder="1"
```

Comment:
I think that the number placeholder being 1 is confusing. I would change this to "add step number"